### PR TITLE
Layout student entries ui router

### PR DESCRIPTION
**Pull request by nmohamed**
Let's Go!

### CODE REVIEW

**Comment by BrennaManning** [comment]
What determines whether an html page is partial-page.html or not? It seems like some are and some aren't. (Just never seen this before, so I'm assuming it's a ui-router thing)

nora: it's just a naming convention; we can change it if it's confusing. the ones that are considered partial are just html files that are nested within other html files
i.e., theres `main.html` which has all your scripts and an empty `<body />`
there's `index.html` which is injected into the `<body />` tag of `main.html`
there's `partial-whatever.html` which is injected into the `<ui-view />` tags in `index.html`

**Comment by BrennaManning** [comment]
Also, there was a comment about .success and .error being deprecated? 
Can whoever wrote that give a little more info - Sounds like we shouldn't be using them. Will they stop working soon? Is there an alternative that exists in a similar format?

nora: they still work, but if i remember what david said correctly, you should use `.then, function(){}` format (see login controller's http.get/post requests for an example). it's not too hard to change

**Comment by BrennaManning** [comment]
Is it necessary to set some of the $scope variables to strings in the controllers? (Lines 139-145 of ng-apps.js)

nora: `ng-apps.js` doesn't have more than 86 lines?

**Comment by BrennaManning** [comment]
Some of the empty-ish html files are a little strange.  The add student from for example already exists. I'm assuming that's just a to-do for later? Which is fine - just clarifying!  

nora: yup, they're just template files for us to move code around eventually

**Comment by BrennaManning** [comment]
line 88 of sutdent.html has a line that is "+>>>>>>> master"  I'm deleting this line now.

**Comment by BrennaManning** [comment]
Had a few comments/questions about some very little things but overall looks good! :+1:

**Comment by nmohamed** [comment]
if commenting on certain lines, make sure to comment on them instead of doing comments here (so it's easier to reference and reply to)

File: public/javascripts/ng-apps.js [hunk 1]
```diff
@@ -1,17 +1,86 @@
...
     })
-    .when('/', {
-      templateUrl: 'views/login.html',
-      controller: 'loginCtrl'
+
+    .state('index', {
+      url: '/',
+      views: {
+        '': {
+            templateUrl: 'views/index.html',
```
Comment (aconsilvio) [inline public/javascripts/ng-apps.js:20]: Does index.html still exist?  it looks like you changed it to main.html

File: public/javascripts/ng-controllers.js [hunk 2]
```diff
@@ -97,6 +97,7 @@ springInitiative.controller('mainController', function($scope, $http) {
       })
   };
 
+  // TODO: .success and .error are deprecated
```
Comment (aconsilvio) [inline public/javascripts/ng-controllers.js:116]: Can someone explain this - I've never heard this.  Also, if that is the case, what is the best way to fix it?

**Comment by nmohamed** [comment]
http://stackoverflow.com/questions/14319967/angularjs-routing-without-the-hash
on the topic of having your route be `website.com/#/page` vs. `website.com/page`

File: public/javascripts/ng-apps.js (hunk 1 above)
Comment (nmohamed) [inline public/javascripts/ng-apps.js:20]: yup it still does - `main.html` and `index.html` are different

File: public/javascripts/ng-controllers.js (hunk 2 above)
Comment (nmohamed) [inline public/javascripts/ng-controllers.js:116]: see my edit on brenna's comment above :arrow_up:

File: public/javascripts/ng-controllers.js [hunk 3]
```diff
@@ -108,3 +109,44 @@ springInitiative.controller('mainController', function($scope, $http) {
...
+springInitiative.controller('studentController', function($scope, $rootScope, $http, $location){
+    $scope.studentName = 'd';
+    $scope.studentInfo = 'we';
+});
+
+springInitiative.controller('addStudentController', function($scope, $rootScope, $http, $location){
+});
+
+springInitiative.controller('settingsController', function($scope, $rootScope, $http, $location){
+});
```
Comment (aconsilvio) [inline public/javascripts/ng-controllers.js]: What is all of this code/where did all of the controller for all of the student front-end go?  (We already have working controller code that handles add students, editing, etc.)  Also, why are we using a new controller for each functions?  Isn't that kind of redundant since this will all be happening on the same page?  I might be totally wrong here, but it seems like the code brenna and i worked on to get angular and the back end working with all students has been removed?  Idk, let me know or maybe it would help it someone could explain this to me tonight.

File: public/views/addStudent.html [hunk 4]
```diff
@@ -0,0 +1 @@
+Add a student using our invisible form:
```
Comment (aconsilvio) [inline public/views/addStudent.html]: There should be a form already created for this.

File: public/views/settings.html [hunk 5]
```diff
@@ -0,0 +1 @@
+Some settings go here
```
Comment (aconsilvio) [inline public/views/settings.html]: Is settings like admin setting?  Just trying to clarify what the purpose of this page will be.

File: public/javascripts/ng-controllers.js (hunk 3 above)
Comment (nmohamed) [inline public/javascripts/ng-controllers.js]: it's in this file under `mainController`

File: public/javascripts/ng-apps.js (hunk 1 above)
Comment (aconsilvio) [inline public/javascripts/ng-apps.js:20]: Ok cool, could you explain the difference between them now then? I thought index was like the the template file for the ngroutes, but what does main do now?

File: public/javascripts/ng-controllers.js (hunk 3 above)
Comment (nmohamed) [inline public/javascripts/ng-controllers.js]: also, although we haven't moved things around yet, adding students is its own separate page, which means it needs its own controller

File: public/views/addStudent.html (hunk 4 above)
Comment (nmohamed) [inline public/views/addStudent.html]: we haven't moved everything around yet. this is just a template file where the add student form will be moved to

File: public/views/settings.html (hunk 5 above)
Comment (nmohamed) [inline public/views/settings.html]: yup - if youre an admin, you'll have admin settings showing up. if youre staff, it'll just be to change some stuff on your account (at the very least, changing your password)

**Comment by aconsilvio** [comment]
Overall it looks good, should we fix the stuff that is split between mainController and the new js file before we push?  I feel weird pushing it to master with so much of that code not integrated, but otherwise i think it looks good.

Also, I can't run this because my git doesn't seem to think that is branch exists for some reason (yes I pulled), so I would like to see it running before we merge.

File: public/javascripts/ng-apps.js (hunk 1 above)
Comment (nmohamed) [inline public/javascripts/ng-apps.js:20]: `main.html` includes everything you'd expect to see in a blank webpage - scripts, the page title, head and body tags, etc. however, the `<body />` tags only hold one div, `<div ui-view />`. this is the equivalent of `<div ng-view />`
`index.html` is code that's injected into `<div ui-view />`. so is `login.html`.
pages such as `student.html` are then injected into `index.html`

**Comment by nmohamed** [comment]
yeah i think we were planning on fixing things this weekend, but we made this pull request to help everyone understand what's going on as we fix things

File: public/javascripts/ng-controllers.js (hunk 3 above)
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js]: @aconsilvio the controller code you wrote is starting at line 75 in the `ng-controllers` file.

File: public/javascripts/ng-controllers.js (hunk 2 above)
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js:116]: @BrennaManning you should use the `.then` function. implementation is very similar. check these [docs](https://docs.angularjs.org/api/ng/service/$http)

Comment (aconsilvio) [inline public/javascripts/ng-controllers.js:116]: :+1:

File: public/javascripts/ng-controllers.js (hunk 3 above)
Comment (aconsilvio) [inline public/javascripts/ng-controllers.js]: :+1:

File: public/javascripts/ng-controllers.js (hunk 2 above)
Comment (davidabrahams) [inline public/javascripts/ng-controllers.js:116]: I will change this! :)

**Comment by davidabrahams** [comment]
@nmohamed Did you ever get the user to display properly on the homepage? It's not working for me.

**Comment by davidabrahams** [comment]
Hey guys, I accidentally forgot to pull and so I renamed the partial-login file (which someone else had already done). Then I had to deal with merge conflicts.

TLDR: my last two commits actually end up not changing anything at all. my bad.

**Comment by davidabrahams** [comment]
@nmohamed I used the code [here](https://scotch.io/tutorials/pretty-urls-in-angularjs-removing-the-hashtag) to remove the hashtags. It's really clean and only 2 lines of code!

**Comment by nmohamed** [comment]
@davidabrahams cool - and np with the commits. currently i'm working on getting students to show properly and dividing `student.html` into the proper controllers. i'll make a new branch and push now, since i got the forms on their own page (still need to refresh to show new student though, but i'll get to it eventually)

branch is called `dividing-student-html`

File: public/javascripts/ng-controllers.js [hunk 6]
```diff
@@ -40,7 +40,30 @@ springInitiative.controller('loginCtrl', function($scope, $http, $location) {
...
+    console.log('Error: in GET \'/student\'', err);
+  });
+});
+
+springInitiative.controller('overviewController', function($scope, $rootScope, $http, $location) {});
+
+springInitiative.controller('programController', function($scope, $rootScope, $http, $location) {
+  $scope.programName = 'something spring';
+  $scope.programInfo = ':3';
+});
```
Comment (shrutiyer) [inline public/javascripts/ng-controllers.js:67]: Are the above lines just placeholders for now?

Comment (davidabrahams) [inline public/javascripts/ng-controllers.js:67]: Yes. I think Nora is fixing it in a sub-branch

File: public/views/content/student.html [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      <br>
+      </input>
+      Archived: <input type="radio" ng-model = "currentStudent.archived" ng-init="currentStudent.archived = currentStudent.archived" name="radAnswer" value = "true"> Yes </input><input type="radio" ng-model = "currentStudent.archived" name="radAnswer" value = "false"> No </input>
+      <button class="btn btn-lg btn-primary btn-block" ng-click = "editStudentFunc(currentStudent)" type="submit">Submit</button>
+    </center>
+  </form>
+</div>
+</div>
+<div class="newStudentForm">
+<form class="studentform">
```
Comment (shrutiyer) [inline public/views/content/student.html]: Is the convention to use singlequotes for js and double for html?

File: views/main.html [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.5/angular.min.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-router/0.2.18/angular-ui-router.js"></script>
+
+    <!-- Our scripts -->
+    <script src="/javascripts/ng-apps.js"></script>
+    <script src="/javascripts/ng-controllers.js"></script>
+
+    <base href="/">
+
```
Comment (shrutiyer) [inline views/main.html:27]: Does it not wrk without base href?

**Comment by shrutiyer** [comment]
How close are we to merging it?

File: public/views/content/student.html (hunk 7 above)
Comment (nmohamed) [inline public/views/content/student.html]: i think so, that's what i've been doing

File: views/main.html (hunk 8 above)
Comment (nmohamed) [inline views/main.html:27]: it's to get rid of the `#` in the URL

**Comment by davidabrahams** [comment]
:+1: MERGE DAT BUS

**Comment by BrennaManning** [comment]
:+1:

**Comment by shrutiyer** [comment]
:+1:
